### PR TITLE
Support for snowflake private key

### PR DESCRIPTION
**Pull request by evgenyx00**

### CODE REVIEW

File: universal_data_permissions_scanner/cli.py [hunk 1]
```diff
@@ -59,14 +59,35 @@ def main(ctx: click.Context, debug: bool, out: str, out_format: str):
...
+@click.option('--rsa-key', '-r', required=False, type=click.File('r'), help="Path to RSA private key")
+@click.option('--rsa-pass', required=False, type=str, help="RSA key password")
 @click.option('--account', '-a', required=True, type=str, help="Account")
 @click.option('--host', '-t', required=False, type=str, help="Hostname")
 @click.option('--warehouse', '-w', required=False, type=str, help="Warehouse")
-def snowflake(ctx: click.Context, username: str, password: str, account: str, host: str, warehouse: str):
+def snowflake(
+    ctx: click.Context,
+    username: str,
+    password: str,
```
Comment (omer-shtivi) [inline universal_data_permissions_scanner/cli.py:71]: I think it also change the password to `Optional[str]` no?

File: universal_data_permissions_scanner/datastores/snowflake/analyzer.py [hunk 2]
```diff
@@ -60,7 +62,7 @@ class SnowflakeAuthzAnalyzer:
     logger: Logger
 
     @classmethod
-    def connect(
+    def connect(  # pylint: disable=too-many-locals
```
Comment (omer-shtivi) [inline universal_data_permissions_scanner/datastores/snowflake/analyzer.py:65]: The password is now optional no?

File: universal_data_permissions_scanner/datastores/snowflake/analyzer.py [hunk 3]
```diff
@@ -92,10 +96,14 @@ def connect(
         # Handle case sensitive warehouse name, wrap with quotes
         warehouse = f'"{warehouse}"'
 
+        if rsa_key is not None:
```
Comment (omer-shtivi) [inline universal_data_permissions_scanner/datastores/snowflake/analyzer.py:99]: Why do we pass some in kwargs an other as field ?

File: universal_data_permissions_scanner/cli.py (hunk 1 above)
Comment (evgenyx00) [reply universal_data_permissions_scanner/cli.py]: yes, changing

File: universal_data_permissions_scanner/datastores/snowflake/analyzer.py (hunk 2 above)
Comment (evgenyx00) [reply universal_data_permissions_scanner/datastores/snowflake/analyzer.py:65]: yes

File: universal_data_permissions_scanner/datastores/snowflake/analyzer.py (hunk 3 above)
Comment (evgenyx00) [reply universal_data_permissions_scanner/datastores/snowflake/analyzer.py:99]: Do you see a problem with that?

Comment (omer-shtivi) [reply universal_data_permissions_scanner/datastores/snowflake/analyzer.py:99]: It's weird, like what is the decision point?

**Review by omer-shtivi — APPROVED** [review]

**Pull request merged**